src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -0,0 +1,109 @@
+from pyspiffe.bundle.x509_bundle.x509_bundle_set import X509BundleSet
+from pyspiffe.bundle.jwt_bundle.jwt_bundle_set import JwtBundleSet
+from pyspiffe.config import ConfigSetter, Config
+from pyspiffe.svid.x509_svid import X509Svid
+from pyspiffe.svid.jwt_svid import JwtSvid
+
+from pyspiffe.workloadapi.workload_api_client import WorkloadApiClient
+
+
+class DefaultWorkloadApiClient(WorkloadApiClient):
+    """
+    Default implementation for a SPIFFE Workload API Client.
+    """
+
+    def __init__(self, spiffe_socket: str = None):
+        """
+        Create a new Workload API Client.
+
+        Args:
+            spiffe_socket (str, optional): Path to Workload API UDS. If
+                not specified, the SPIFFE_ENDPOINT_SOCKET environment variable
+                must be set.
+
+        Returns:
+            DefaultWorkloadApiClient: New Workload API Client object.
+
+        Raises:
+            ValueError if spiffe_socket_path is invalid or not provided and
+                SPIFFE_ENDPOINT_SOCKET environment variable doesn't exist
+        """
+
+        try:
+            self._config = ConfigSetter(
+                spiffe_endpoint_socket=spiffe_socket
+            ).get_config()
+        except ValueError:
+            raise ValueError(
+                'SPIFFE socket argument or environment variable invalid in DefaultWorkloadApiClient'
+            )
+
+    def get_config(self) -> Config:
```

Comment:
It's probably not such a good idea to expose the Config object, what would the use of this `get` method ?
In case you decide to keep it, use `config(self)`

Reply:
Resolved in latest commit, with get_config() replaced by get_spiffe_endpoint_socket()